[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/gamayos/finite-ring-space/blob/main/src/21-gravity/21-gravity-main.ipynb)

## Gravitation as Phase Synchronisation over Finite Holographic Substrate (Akhtman & Voether, 2026)

**Validation Package.** `finite-ring-space/src/21-gravity` — the paper's twenty-one validation scripts as written, run through one registry: a
family check is one script's verdict (`grav.<stem>`), its micro-checks the script's labelled checks and every PASS/FAIL line it
prints; each family names the row(s) of the paper's predicate ledger it witnesses (the paper's Section "Status", subsection
"Predicate ledger", rows cited as `21:XN`; public copy at `docs/21-gravity/21-gravity-ledger.html`), and the ledger's source column
cites the family ids in return. Where a row is proved in Lean (`lean/FrcCore/Gravity.lean` with no axioms, or
`lean/FrcLedger/Gravity.lean` on Mathlib — the Carrier register, the count face and the merger law, the cover forcing, the operational
cut and the photon sphere, the registration root and the crossover's limits, the shift theorem, with the shadow, ringdown, efficiency,
floor, tilt and interpolant numerals bracketed), the check here is the instance the reader can run. Master-ledger rows of the corpus
reached through the paper rows: `00:A9`, `00:D1`, `00:D3`, `00:D5`, `00:B7`, `00:E1`, `00:D15`, `00:E3`–`00:E7`, `00:E10`, `00:L1`,
`00:L7`, `00:L8`, `00:T1`, `00:T4`, `00:T5`, `00:Z7`.

**Kinds.** `EXACT`: decidable identities over finite fields, exact rationals or symbolic identities, no tolerance in the verdict.
`CHART`: a continuum reading or a comparison with measured data, tagged [approx]/[data] in the script, verdict by stated tolerance.
`SIM`: a seeded stochastic simulation, verdict by stated tolerance on time averages.

**Run.** Cell by cell, or *Runtime → Run all*; ≈ 1–2 minutes on Colab. The last cell writes `results.json` and fails loudly if any family fails.

In [1]:
# --- environment: clone the package if this notebook is not already running inside it (Colab). numpy and sympy are present on Colab.
import os
if not os.path.exists("gravcommon.py"):
    if not os.path.exists("finite-ring-space"):
        !git clone --filter=blob:none --sparse https://github.com/gamayos/finite-ring-space.git
        os.chdir("finite-ring-space")
        !git sparse-checkout set src/21-gravity
        os.chdir("src/21-gravity")
    else:
        os.chdir("finite-ring-space/src/21-gravity")
print("working directory:", os.path.join(*os.getcwd().split(os.sep)[-3:]))

working directory: frs-public/src/21-gravity


In [2]:
import importlib, gravcommon as gc
importlib.reload(gc)         # a fresh registry if the notebook is re-run
print("families registered:", len(gc.LABELS))

families registered: 21


## The Newtonian and relativistic layers
The lattice Green's function and the inverse-square law; the PPN and classical-test factors; the nonlinear completion (cut-flux law, the exponential reading unique); the discrete Fierz–Pauli gauge invariance and its uniqueness; the radiative sector; the order-one constants and the Carrier-register residues.

| id | script | kind | claim | ledger rows |
|---|---|---|---|---|
| `grav.newton` | `validate_newton.py` | CHART | the Newtonian layer: the exact lattice Green's function of Z³ (Montroll–Bessel) with 4πr·G(r) → 1 and the O(r⁻²) correction, the r⁻¹ potential (r⁻² force), the locked ensemble coupling as m against the incoherent √m; the spanning-tree admissibility det L_red = τ(C_n) = n (n = 5, 8, 12), the finite solvability predicate | 21:C1, 21:C2 |
| `grav.ppn` | `validate_ppn.py` | EXACT | the relativistic layer: Fermat deflection 4Gm/c²b in the index n = 1 + 2u; β = γ = 1 from the exponential metric; the perihelion factor (2 + 2γ − β)/3 = 1; the static-profile series Gm/r·(1 + (Gm/r²)²/30) — symbolic identities | 21:C7, 21:C8, 21:C12 |
| `grav.branch` | `validate_branch.py` | CHART | the nonlinear completion: the cut-flux law of the full sine model (transfer antisymmetry) — flux through every closed surface equals the enclosed demand to solver precision for one and two sources at strong gradients, the nonlinear deviation scaling as m³ (no m² self-sourcing); symbolically, isotropic Schwarzschild is the exponential reading of ψ = 2 artanh(U/2), ψ non-harmonic, and the Schwarzschild reading violates the composition law R(a+b) = R(a)R(b) the exponential uniquely satisfies | 21:C11 |
| `grav.fp_gauge` | `validate_fp_gauge.py` | EXACT | the discrete Fierz–Pauli functional on a periodic (3+1) lattice with integer random fields: gauge invariance under h → h + Δξ an exact integer identity for central differences and failing for one-sided differences (the control) — anti-self-adjointness Δᵀ = −Δ carries the proof; Schwartz–Zippel on random integer fields | 21:C9 |
| `grav.fierz_pauli` | `fierz_pauli_uniqueness.py` | EXACT | uniqueness: by the symbol/transversality method the discrete Fierz–Pauli functional is the unique adjacency-local gauge functional on the shell in the (3+1) signature — the rank-2 analogue of Maxwell uniqueness | 21:C9 |
| `grav.radiative` | `validate_radiative.py` | EXACT | the radiative sector: the quarter-turn conjugate momentum makes the dynamics Hamiltonian, χü = κΔu; channel unity gives c_g² = κ/χ = c²; the Fierz–Pauli field carries two transverse-traceless polarisations of helicity ±2; the dispersion relation, the long-wave speed, TT rank 2, no mode doubling (Nyquist); the symbol even in k, on the continuum form and on C₁₃ | 21:C15 |
| `grav.orderone` | `validate_orderone.py` | EXACT | the order-one constants: c_S′ = 1/4 from the de Sitter closure (the same area law on the cosmological horizon √Ω reproduces the de Sitter entropy defining the cardinality); the 2π of a₀ as the angular period of one phase turn; the radiation constant as the spin-2 quadrupole factor; the 4π; in exact rationals — and on the instantiated Carrier Ω = 2 408 561 the register residues (4S)² ≡ 1, G = 2S = (2·4S)⁻¹, G = −c², c² = 2⁻¹, ħ² = −1, G_p = π_p on four shells | 21:C5, 21:C16, 21:C13 |

Ledger rows witnessed: 21:C1, 21:C2, 21:C5, 21:C7, 21:C8, 21:C9, 21:C11, 21:C12, 21:C13, 21:C15, 21:C16.

In [3]:
for fam in ["grav.newton", "grav.ppn", "grav.branch", "grav.fp_gauge", "grav.fierz_pauli", "grav.radiative", "grav.orderone"]:
    print(f"\n— {gc.script_of(fam)}.py"); gc.run_block(fam)


— validate_newton.py


  exact lattice Green vs 1/(4 pi r):


    r= 3: 4*pi*r*G(r) = 1.03843
    r= 5: 4*pi*r*G(r) = 1.01166
    r= 8: 4*pi*r*G(r) = 1.00411
    r=12: 4*pi*r*G(r) = 1.00177
  potential power law: U ~ r^-1.0113  (target -1)
  coupling: locked = 10000 (=m), unlocked ~ 90 (target ~ sqrt(pi m)/2 = 89)
PASS - Newtonian layer
PASS spanning-tree admissibility: det L_red = tau(C_n) = n exactly (n = 5, 8, 12); tau != 0 mod Omega is the finite solvability predicate
  [PASS] grav.newton            CHART [21:C1, 21:C2] the Newtonian layer: the exact lattice Green's function of Z³ (Montroll–Bessel) with 4πr·G  --  2 checks, 1.4 s

— validate_ppn.py


  deflection: 4*G*m/(b*c**2)  (target 4Gm/(b c^2))
  PPN: beta = 1, gamma = 1  (targets 1, 1)
  perihelion factor (2+2*gamma-beta)/3 = 1  (target 1)


  static profile: Gm**3/(30*r**5) + Gm/r  (target Gm/r + Gm^3/(30 r^5))
PASS - relativistic layer / PPN
  [PASS] grav.ppn               EXACT [21:C7, 21:C8, 21:C12] the relativistic layer: Fermat deflection 4Gm/c²b in the index n = 1 + 2u; β = γ = 1 from   --  1 checks, 0.8 s

— validate_branch.py


  (A) single source m=4.0, solver residual 2.3e-14
      flux through box R=2: 4.0000000000  (demand 4.0)
      flux through box R=4: 4.0000000000  (demand 4.0)
      flux through box R=6: 4.0000000000  (demand 4.0)
      flux through box R=9: 3.9999999999  (demand 4.0)


      two sources, enclosing box: flux = 5.9999999998  (demand 6.0)


      nonlinear deviation scaling: ||u_sin - u_lin|| ~ m^3.06  (target 3: cubic, no m^2 term)
  (B) g00(Schwarzschild isotropic) == exp(-2 psi), psi = 2 artanh(U/2): residual 0
      Laplacian(psi) = Gm**3/(2*r**5)  (nonzero: psi is self-sourced; u = Gm/r is harmonic)


      composition law: exponential satisfies R(a+b)=R(a)R(b): True; Schwarzschild reading violates it (mismatch -2*a**2*b/(a**2*b + 2*a**2 + a*b**2 + 6*a*b + 8*a + 2*b**2 + 8*b + 8) - 2*a*b**2/(a**2*b + 2*a**2 + a*b**2 + 6*a*b + 8*a + 2*b**2 + 8*b + 8) != 0)
PASS - branch decision (Ward identity + composition law)
  [PASS] grav.branch            CHART [21:C11] the nonlinear completion: the cut-flux law of the full sine model (transfer antisymmetry)   --  1 checks, 42.2 s

— validate_fp_gauge.py
  central  differences: gauge invariance exact (integer identity, D^T = -D)
  forward  differences: gauge invariance broken (E1 - E0 = -6296)

PASS - discrete (3+1) FP gauge invariance: exact integer identity for central differences
  [PASS] grav.fp_gauge          EXACT [21:C9] the discrete Fierz–Pauli functional on a periodic (3+1) lattice with integer random fields  --  1 checks, 0.0 s

— fierz_pauli_uniqueness.py
SPIN-2  continuum symbol k_mu


  [PASS] gauge-invariant 2-deriv quadratic space is 1-dimensional
  [PASS] basis = (-1,2,-2,1) = linearized Einstein-Hilbert (Fierz-Pauli)
SPIN-2  on the shell: central-difference symbol s_mu = sin k_mu


  [PASS] uniqueness holds on the shell (1-dimensional)
  [PASS] same solution = discrete Fierz-Pauli functional eq.(dfp)
Central vs forward difference: anti-self-adjointness on cyclic Z_L
  [PASS] central difference is anti-self-adjoint (D^T = -D), real symbol sin k
  [PASS] forward difference is NOT (so its FP functional is not gauge invariant)
Fierz-Pauli MASS term: unique ghost-free tuning
  [PASS] coeff of h00^2 in [b1 h_mn h^mn + b2 h^2] = b1 + b2
  [PASS] vanishes iff b1+b2=0  => mass term ∝ (h_mn h^mn - h^2), the FP combination
SPIN-1  Maxwell cross-check (same method, reproduces Thm o1)
  [PASS] Maxwell: gauge-invariant relevant operator unique (1-dimensional)
  [PASS] basis ∝ (1,-1) = F_mn F^mn = 2[k^2 A^2 - (k.A)^2]
RESULT: 10/10 checks pass
Generic adjacency-local uniqueness lemma, all three instances on the shell:
  spin-1 Maxwell, spin-2 Fierz-Pauli, Yang-Mills (per-colour spin-1 +
  unique Killing form).  The relevant gauge-invariant quadratic functional
  is the unique c

PASS symbol check: dispersion even in k (odd coefficients 0,0,0,0) -- no O(k).n cross term
  [PASS] grav.radiative         EXACT [21:C15] the radiative sector: the quarter-turn conjugate momentum makes the dynamics Hamiltonian,   --  2 checks, 0.3 s

— validate_orderone.py
[1] c_S' fixed to the Bekenstein-Hawking 1/4:
    operational-horizon ratio S_op/S_BH=(r_f/r_s)^2  =>  4 c_S' = 1  =>  c_S' = 1/4
    de Sitter closure  c_S' * 4 pi Omega = pi Omega  =>  c_S' = 1/4
    both give c_S' = 1/4: fixed, and it reproduces Bekenstein-Hawking-Gibbons-Hawking.

[2] the recurring geometric constant is the solid angle 4 pi:
    G = 1/(4 pi kappa), alpha_bare = 1/4 pi, Green's function ~ 1/(4 pi r), area = 4 pi r^2.

[3] a0 = c H0 / 2 pi, the 2 pi = the phase-cycle period (one full turn) [approx]:
    H0= 67.4: a0 = 1.042e-10 m/s^2
    H0= 70.0: a0 = 1.082e-10 m/s^2
    H0= 73.0: a0 = 1.129e-10 m/s^2
    within ~10% of the fitted 1.20e-10; the 2 pi is the cycle period, not a free coefficient.

[4

## The strong field and the rotating solution
The static profile with its photon sphere, shadow, ISCO and accretion efficiency; frame dragging as the quarter-turn dual; the fold-return criterion at dev scale (no prompt echo).

| id | script | kind | claim | ledger rows |
|---|---|---|---|---|
| `grav.strongfield` | `validate_strongfield.py` | CHART | the strong field: the slip-core boundary r* = √(Gm); the photon sphere r_ph = 2Gm with b_c = 2e·Gm (shadow +4.6 %, ringdown −4.4 %); the ISCO at r = (3 + √5)Gm with accretion efficiency 5.48 % against Schwarzschild's 5.72 % and Gm·Ω/c³ = 0.0633 (0.931×); r_f = r_s/ln Ω, S/S_BH = ln⁻²Ω, the echo delay factor Ω/ln²Ω; Sgr A*/M87* angular diameters 53.3 → 55.7 µas, 39.7 → 41.5 µas [approx] | 21:C12, 21:C13, 21:P1, 21:P7 |
| `grav.rotating` | `validate_rotating.py` | CHART | the rotating solution as the quarter-turn dual of the static field: the Lense–Thirring frame-dragging ω = 2GJ/c²r³, the gravitomagnetic dipole, horizonless-ness (the redshift floor only), the O(a) shadow spin shift (structure only) | 21:C10, 21:C17 |
| `grav.fold_echo` | `validate_fold_echo.py` | EXACT | the fold-return criterion at dev scale, exact over F₁₇: on the chain C₈ with the interior block folded by its exact DFT, the period-averaged exterior return P_E F_I Uⁿ F_I⁻¹ P_E equals the rank-one uniform-floor operator N⁻¹J_E, fold-independent (the fold drops out of the average; master E10) | 21:C12, 21:P7 |

Ledger rows witnessed: 21:C10, 21:C12, 21:C13, 21:C17, 21:P1, 21:P7.

In [4]:
for fam in ["grav.strongfield", "grav.rotating", "grav.fold_echo"]:
    print(f"\n— {gc.script_of(fam)}.py"); gc.run_block(fam)


— validate_strongfield.py
  slip core r_* = sqrt(Gm) = 1000  (arcsin domain boundary)
  u(2Gm) = 0.499999  vs Gm/r = 0.5  (exact profile ~ Newtonian at photon sphere)


  photon sphere r_ph = 2.000 Gm (target 2),  b_c = 5.4366 Gm (target 2e = 5.4366)
  shadow deviation delta = 4.63%  (target +4.63), ringdown -4.42% (target -4.42)
  r_f = 2Gm/ln(Omega) = r_s/281,  S/S_BH = 1.27e-05,  echo delay factor Omega/ln^2(Omega) = 1.3e+117 (no observable echoes)
  Sgr A*: shadow GR 53.3 muas, FRC 55.7 muas
  M87*: shadow GR 39.7 muas, FRC 41.5 muas
  ISCO r = (3+sqrt5) Gm = 5.2361 Gm,  |W''| = 0.0e+00 (marginally stable)
  E/m = 0.94521,  efficiency = 5.479% (Schwarzschild 5.719%),  Gm*Omega = 0.06333, ratio to Schwarzschild = 0.9308
PASS - strong field
  [PASS] grav.strongfield       CHART [21:C12, 21:C13, 21:P1, 21:P7] the strong field: the slip-core boundary r* = √(Gm); the photon sphere r_ph = 2Gm with b_c  --  1 checks, 2.0 s

— validate_rotating.py
[1] frame-dragging omega(r) = 2J/r^3 (Lense-Thirring):
    a=J/M=0.1: omega(10M)=2.000e-04 = 2GJ/c^2 r^3  [recovered]
    a=J/M=0.3: omega(10M)=6.000e-04 = 2GJ/c^2 r^3  [recovered]
    a=J/M=0.6: omega(10M)=1.20

## The floor, the crossover and registration
The acceleration floor against SPARC; the registration crossover and its orbital response; the two-shift update law; the projection defect; the unified counting checks; flux conservation under drive noise.

| id | script | kind | claim | ledger rows |
|---|---|---|---|---|
| `grav.rar` | `validate_rar.py` | CHART | the weak-acceleration floor a₀ = cH₀/2π for H₀ ∈ {67.4, 70, 73} against the SPARC 1.20 ± 0.24 × 10⁻¹⁰ m s⁻² (all within one systematic σ); the Local-Group turnaround radius (Gm/H²)^{1/3} ≈ 1.6 Mpc against the observed ~1 Mpc [data] | 21:C4, 21:P3 |
| `grav.deepregime` | `validate_deepregime.py` | EXACT | the registration crossover: the floor as link-decorrelation rate, x = g_b/a₀ cycles per Hubble time with Born amplitude √x; the resolved fraction f = 1 − η^a with η the in-(0,1) root of the rational quadratic wη² − 2η + w = 0; the deep-MOND limit g_obs → √(g_b a₀) (BTFR), the Newtonian limit f → 1, the η root exact; the e^{−√x} reading [approx] | 21:C19, 21:P4 |
| `grav.deepregime_orbit` | `validate_deepregime_orbit.py` | EXACT | the registered-inertia identity and the two-boundary law: the Chebyshev masking weight q_{L,d} against the direct tridiagonal solve of the killed recurrence, exact-rational, on three (L, d, w) triples; impulse in = m g T = registered momentum change identically for framed-rational f = N/T; the event-based killed-registration orbit — registered circular balance v²/r = g_N/f as an exact identity of the event tallies, the offset-sector ledger closing the window total exactly | 21:C19, 21:C20 |
| `grav.inertia` | `validate_inertia.py` | EXACT | the two-shift update law: the kick as the finite Fourier shift theorem in F_p (exact, no error term); the Ehrenfest parabola on exact chirps, Δ²⟨q⟩ = −s/m as an exact integer second difference; the equivalence principle bitwise — the runs (m, F) and (3m, 3F) give identical registered trajectories | 21:C20 |
| `grav.defect` | `validate_defect.py` | EXACT | the projection defect on dev shells: cycle-quotient closure exact (defect 1); windowed registration accumulating exactly the wrap count; K_S = P U_C P contractive on a non-invariant sector while U_C is a permutation of finite order; on the odd-character space of C₁₂ the antisymmetric currents are spanned by odd harmonics with the sine mode the unique minimal degree — the minimal-degree condition a declared realisation | 21:C21 |
| `grav.counting` | `validate_counting.py` | EXACT | unified counting in exact integers: the Christoffel registration pattern (a advances per recurrence b, gaps ⌊b/a⌋, ⌈b/a⌉); the censored geometric identity symbolically; the readout instance Ω = 641, g = 3 (s₄ = 114, R₄ = 23/54 and 2/27, the largest-remainder allocation 35 + 34 + 6 + 6 = 81); the negative control at B = 108 (Kesten); the per-stratum split at B = 2, 3 as Galois-symmetric multinomials of (W₊ + W₋)^B in Z[√2] and mod 641 | 21:C7, 21:C21 |
| `grav.fluxnoise` | `validate_fluxnoise.py` | SIM | flux conservation under noise: a driven, pinned Kuramoto chain in the statistically stationary regime — the time-averaged transport through every link equals the source demand with and without drive noise (seeded simulation) | 21:C19 |

Ledger rows witnessed: 21:C4, 21:C7, 21:C19, 21:C20, 21:C21, 21:P3, 21:P4.

In [5]:
for fam in ["grav.rar", "grav.deepregime", "grav.deepregime_orbit", "grav.inertia", "grav.defect", "grav.counting", "grav.fluxnoise"]:
    print(f"\n— {gc.script_of(fam)}.py"); gc.run_block(fam)


— validate_rar.py
  a0 = c H0 / (2 pi):
    H0= 67.4: a0 = 1.042e-10 m/s^2   (-0.66 syst-sigma from fit)
    H0= 70.0: a0 = 1.082e-10 m/s^2   (-0.49 syst-sigma from fit)
    H0= 73.0: a0 = 1.129e-10 m/s^2   (-0.30 syst-sigma from fit)
  Local Group turnaround (Gm/H^2)^(1/3) = 1.64 Mpc  (observed zero-velocity ~ 1 Mpc)
PASS - weak-acceleration floor
  [PASS] grav.rar               CHART [21:C4, 21:P3] the weak-acceleration floor a₀ = cH₀/2π for H₀ ∈ {67.4, 70, 73} against the SPARC 1.20 ± 0  --  1 checks, 0.0 s

— validate_deepregime.py
[1] First-passage registration: exact algebraic eta vs [approx] e^{-sqrt x}
    s=0.020000 a= 3  eta^a=0.546039  [approx]e^(-sqrt x)=0.548812  sqrt(x)=0.600  rel=5.05e-03
    s=0.005000 a= 6  eta^a=0.548124  [approx]e^(-sqrt x)=0.548812  sqrt(x)=0.600  rel=1.25e-03
    s=0.001250 a=12  eta^a=0.548640  [approx]e^(-sqrt x)=0.548812  sqrt(x)=0.600  rel=3.13e-04
    s=0.000313 a=24  eta^a=0.548769  [approx]e^(-sqrt x)=0.548812  sqrt(x)=0.600  rel=7.81e-05
 

PASS (g) pair-tally deficit unique on unit-capacity support: E = a1(z + z^{-1} - 2), one scale

ALL PASS
  [PASS] grav.defect            EXACT [21:C21] the projection defect on dev shells: cycle-quotient closure exact (defect 1); windowed reg  --  9 checks, 0.2 s

— validate_counting.py
PASS christoffel a/b=23/54: a regs per recurrence, two-valued gaps
PASS christoffel a/b=2/27: a regs per recurrence, two-valued gaps
PASS christoffel a/b=1/5: a regs per recurrence, two-valued gaps
PASS christoffel a/b=3/8: a regs per recurrence, two-valued gaps
PASS christoffel a/b=7/12: a regs per recurrence, two-valued gaps
PASS censored geometric identity (exact rational, q=1/7, T=11)
PASS s_4 = 114 and R_4 = 23/54
PASS largest-remainder allocation 35+34+6+6 = 81, discrepancy <= 1
PASS negative control: raw-orbit prefix count breaks unit discrepancy (first at B=4)
PASS dial values are +/- sqrt2 mod 641 and sum to zero
PASS B=2: string tallies = |dial| x Galois-symmetric coefficients; total = 4^B x |

  noise=0.0: mean flux per link = 0.3000 (demand b=0.3), max deviation 0.0000


  noise=0.3: mean flux per link = 0.2961 (demand b=0.3), max deviation 0.0123
PASS - flux conservation under noise
  [PASS] grav.fluxnoise         SIM   [21:C19] flux conservation under noise: a driven, pinned Kuramoto chain in the statistically statio  --  1 checks, 13.2 s


## The two-body sector and the primordial spectrum
The 2PN inputs and the J0737−3039 excess; the 1PN equivalence lemma; the binding bookkeeping; the scale-invariant spectrum.

| id | script | kind | claim | ledger rows |
|---|---|---|---|---|
| `grav.2pn` | `validate_2pn.py` | EXACT | the 2PN inputs: the isotropic-gauge deviations of the exponential metric from Schwarzschild, δA = U³/6 in −g₀₀ and δB = U²/2 in g_ij, certified term by term; the test-particle periastron excess π(GM/c²p)²(2 + e²/2) from the matched-invariant orbit equation (δc₂, δc₃ and their apsidal sum); for PSR J0737−3039 the fractional excess ≈ 1.5 × 10⁻⁶ [chart] | 21:C8, 21:P6 |
| `grav.1pn_eih` | `validate_1pn_eih.py` | EXACT | the 1PN equivalence lemma: through first post-Newtonian order the two-body metric coefficients A to O(U²) and B to O(U) coincide with GR's isotropic form on the superposed potential U₁ + U₂, cross term included — the Einstein–Infeld–Hoffmann sector shared exactly; the 2PN ν-sector left open | 21:C23 |
| `grav.binding` | `validate_binding.py` | EXACT | binding bookkeeping decided in exact arithmetic on the torus Z_N³: superposition exact (no O(G²) term); the far-field cut flux around a bound pair equals m₁ + m₂ exactly (substrate winding sources); the interaction energy exactly bilinear; the binding defect m₁(1 − u₂(a)) a registered-clock identity — hence η_Nordtvedt = 0 with the two-shift cancellation | 21:C21, 21:C23 |
| `grav.primordial` | `validate_primordial.py` | EXACT | the primordial spectrum: P ∝ kⁿ scale-invariant iff n = −3 (n_s = 1) uniquely; the observed red tilt n_s − 1 = −0.035 has the predicted sign [data]; wrapped-chart truncation of correlations beyond ~60°; the O2 probe K²λ₁ → π² monotonically (99.1 % at K = 281) | 21:C18, 21:X6, 21:P8 |

Ledger rows witnessed: 21:C8, 21:C18, 21:C21, 21:C23, 21:P6, 21:P8, 21:X6.

In [6]:
for fam in ["grav.2pn", "grav.1pn_eih", "grav.binding", "grav.primordial"]:
    print(f"\n— {gc.script_of(fam)}.py"); gc.run_block(fam)


— validate_2pn.py
PASS delta A = U^3/6 exactly (temporal, first difference at 3PN order in g_00)
PASS delta B = U^2/2 exactly (spatial, first difference at 2PN order in g_ij)
PASS 1PN agreement exact: g_00 and g_ij coincide through O(U^2) and O(U) resp.
    [chart] U_p = 4.381e-06; fractional 2PN excess = 1.46e-06
PASS fractional excess ~1.5e-6 (within [1.0e-6, 2.0e-6])
    verdict: at current 1e-6 omega-dot precision, degenerate with the mass refit;
    decisive once M_tot is fixed independently at 1e-6 (prediction P6).


PASS delta c2 leading term = eps M^2/L^2 exactly
PASS delta c3 = (5/6) M^3/L^2 exactly
PASS apsidal sum identity: 5/2 - (1-e^2)/2 = 2 + e^2/2

ALL PASS
  [PASS] grav.2pn               EXACT [21:C8, 21:P6] the 2PN inputs: the isotropic-gauge deviations of the exponential metric from Schwarzschil  --  8 checks, 0.3 s

— validate_1pn_eih.py
PASS 1PN: -g00 identical through (U1+U2)^2 incl. the cross term 4 U1 U2
PASS 1PN: g_ij identical through (U1+U2)^1
    => EIH two-body sector shared exactly at 1PN; nu-sector at 2PN remains OPEN (O4).

ALL PASS
  [PASS] grav.1pn_eih           EXACT [21:C23] the 1PN equivalence lemma: through first post-Newtonian order the two-body metric coeffici  --  3 checks, 0.2 s

— validate_binding.py


PASS (i) exact superposition u_pair = u1 + u2 on all cells
PASS (ii) far-field cut flux = m1 + m2 = 8 exactly (substrate winding; no defect)
PASS (iii) E[u1+u2] - E[u1] - E[u2] = -<s1,u2> exactly (bilinear, no O(G^2) remainder)
PASS (iv) cluster-1 registered rate in the pair field = m1(1 - u2(a)) (registration-side, linear)

VERDICT: the field is sourced by SUBSTRATE winding (far-field flux = Sum m,
exactly); the binding defect is a REGISTERED-clock effect m(1-u); with the
two-shift m-cancellation, EP holds at O(binding): eta_Nordtvedt = 0.

ALL PASS
  [PASS] grav.binding           EXACT [21:C21, 21:C23] binding bookkeeping decided in exact arithmetic on the torus Z_N³: superposition exact (no  --  5 checks, 2.7 s

— validate_primordial.py
[1] scale-invariance = the dilation fixed point:
    P(k)~k^-2: Delta^2~k^+1  scale-invariant=False  n_s=2
    P(k)~k^-3: Delta^2~k^+0  scale-invariant=True  n_s=1
    P(k)~k^-4: Delta^2~k^-1  scale-invariant=False  n_s=0
    unique invariant: P~k^-3

## Summary
Writes `results.json` (one record per family: id, the ledger row(s) witnessed, script, kind, claim, PASS/FAIL, the micro-check count) and raises if any family failed.

In [7]:
ok = gc.summary(write=True)
failed = [r["id"] for r in gc.RESULTS if not r["ok"]]
assert ok, f"FAILED families: {failed}"
print(f"all {len(gc.RESULTS)} families pass ({len(gc.MICRO)} micro-checks); records in results.json")


SUMMARY: 21/21 checks passed (80 micro-checks)
all 21 families pass (80 micro-checks); records in results.json
